### ISSUE TITLE
`collectSingleFamilyKeys` may return duplicate keys and these duplicates go to the OpenPGP message

### ISSUE THREAD

**Issue by rrrooommmaaa** [FlowCrypt/flowcrypt-browser#4511]
Create a unique "set" based on `KeyUtil.identityEquals`.
Should we do this in `collectSingleFamilyKeys` method or on the `MsgUtil` level?

**Comment by rrrooommmaaa** [comment]
Why does it have S/MIME tag? Duplicate recipients go to OpenPGP message

**Comment by tomholub** [comment]
I added it because it relates to S/MIME in the sense that if no S/MIME keys are imported, this bug won't surface.

**Comment by rrrooommmaaa** [comment]
This bug is related to a situation when several recipients (e.g. aliases) share the same OpenPGP key and several copies of the same key are selected for encryption, get passed to OpenPGP library and even on that level (at least with v4) duplicates aren't thrown away.

**Comment by tomholub** [comment]
Ah. Is that in a situation when I write `to:primary@example.com,alias@example.com` and they have the same key?

**Comment by rrrooommmaaa** [comment]
> Ah. Is that in a situation when I write `to:primary@example.com,alias@example.com` and they have the same key?

Yes, this can be one of the cases. Some other cases may be less evident as my own keys are also added for encryption.
